nifi-nar-bundles/nifi-neo4j-bundle/nifi-neo4j-processors/src/main/java/org/apache/nifi/processors/neo4j/Neo4JCypherExecutor.java

```diff
@@ -0,0 +1,203 @@
+        tempDescriptors.add(PASSWORD);
+        tempDescriptors.add(QUERY);
+        tempDescriptors.add(LOAD_BALANCING_STRATEGY);
+        tempDescriptors.add(CONNECTION_TIMEOUT);
+        tempDescriptors.add(MAX_CONNECTION_POOL_SIZE);
+        tempDescriptors.add(MAX_CONNECTION_ACQUISITION_TIMEOUT);
+        tempDescriptors.add(IDLE_TIME_BEFORE_CONNECTION_TEST);
+        tempDescriptors.add(MAX_CONNECTION_LIFETIME);
+        tempDescriptors.add(ENCRYPTION);
+        tempDescriptors.add(TRUST_STRATEGY);
+        tempDescriptors.add(TRUST_CUSTOM_CA_SIGNED_CERTIFICATES_FILE);
+
+        propertyDescriptors = Collections.unmodifiableList(tempDescriptors);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return propertyDescriptors;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        super.onScheduled(context);
+    }
+
+    @Override
+    protected Collection<ValidationResult> customValidate(ValidationContext validationContext) {
+        List<ValidationResult> results = new ArrayList<>();
+
+        PropertyValue trustStrategy = validationContext.getProperty(AbstractNeo4JCypherExecutor.TRUST_STRATEGY);
+        if (trustStrategy.isSet() && trustStrategy.getValue().equals(AbstractNeo4JCypherExecutor.TRUST_CUSTOM_CA_SIGNED_CERTIFICATES.getValue())) {
+            if ( ! validationContext.getProperty(AbstractNeo4JCypherExecutor.TRUST_CUSTOM_CA_SIGNED_CERTIFICATES_FILE).evaluateAttributeExpressions().isSet() ) {
+                results.add(new ValidationResult.Builder()
+                    .subject(TRUST_STRATEGY.getDisplayName() + " with " + TRUST_CUSTOM_CA_SIGNED_CERTIFICATES.getDisplayName())
+                    .explanation(TRUST_STRATEGY.getDisplayName() + " with " + TRUST_CUSTOM_CA_SIGNED_CERTIFICATES.getDisplayName() + " requires "
+                         + AbstractNeo4JCypherExecutor.TRUST_CUSTOM_CA_SIGNED_CERTIFICATES_FILE.getDisplayName() + " to be set").valid(false).build());
+            }
+        }
+        return results;
+
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        String query = context.getProperty(QUERY).evaluateAttributeExpressions(flowFile).getValue();
+
+        try {
+            long startTimeMillis = System.currentTimeMillis();
+
+            StatementResult statementResult = executeQuery(query);
+
+            List<Map<String, Object>> returnValue = statementResult.list().stream().map(i -> i.asMap()).collect(Collectors.toList());
+
+            getLogger().debug("Result of query {} is {}", new Object [] { query, returnValue });
+
+            Gson gson = new Gson();
```

Comment:
According to [this](https://groups.google.com/forum/#!topic/google-gson/Vju1HuJJUIE), `Gson` is not thread-safe. Jackson is. Switching over to Jackson for this would let you instantiate one ObjectMapper instead of throwing away Gsons every call here.